sparse/impl/KokkosSparse_sptrsv_symbolic_impl.hpp

```diff
@@ -279,6 +281,7 @@ void lower_tri_symbolic(TriSolveHandle& thandle, const RowMapType drow_map,
           check_count);
       std::cout << "  host check_count= " << check_count << std::endl;
 
+      space.fence();    // wait for deep copy writes to land
       check_count = 0;  // reset
       Kokkos::parallel_reduce(
           "check_count device",
```

Comment:
Shouldn't ``ExecSpaceIn`` be used on the next line? The device_t of dnodes_per_level doesn't necessarily match the DefaultExecutionSpace, for example DefaultExecutionSpace could be Cuda but the exec space of the handle and dnodes_per_level is Serial/HostSpace.

Reply:
Good catch

Reply:
@e10harvey I only meant the parallel_reduce starting on line 286 - the one above reads from nodes_per_level which is host. That one can't use ``space`` in general because it might not be able to read host memory